File: go/src/pe009.go
package main

import (
	"fmt"
)

func main() {
	const STOP = 1000
	for a := 0; a < STOP; a++ {
		for b := a + 1; b < STOP; b++ {
			for c := b + 1; c < STOP; c++ {
				if a+b+c == STOP {
					if a*a+b*b == c*c {
						fmt.Printf("%d\n", a*b*c)
						return
					}
				}
			}
		}
	}
}

File: ruby/pe009.rb
#!/usr/bin/ruby

STOP = 1000

(0 ... STOP).each do |a|
  (a+1 ... STOP).each do |b|
    (b+1 ... STOP).each do |c|
      if a + b + c == STOP
        if a*a + b * b == c*c
          puts(a * b * c)
          exit
        end
      end
    end
  end
end
